src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/IONFILELocalFilesHelper.kt

```diff
@@ -0,0 +1,228 @@
+     * @return a (cold) flow in which the chunks are emitted; the flow completes after emissions
+     */
+    @VisibleForTesting
+    internal fun readFileInChunks(
+        fullPath: String,
+        options: IONFILEReadInChunksOptions,
+        bufferSize: Int
+    ): Flow<String> = flow {
+        val file = File(fullPath)
+        if (!file.exists()) {
+            throw IONFILEExceptions.DoesNotExist()
+        }
+        FileInputStream(file).use { inputStream ->
+            inputStream.readByChunks(
+                options,
+                bufferSize,
+                onChunkRead = { chunk -> emit(chunk) }
+            )
+        }
+    }.flowOn(Dispatchers.IO)
+
+    /**
+     * Gets information about a file or directory
+     *
+     * @param fullPath the full path to the file or directory
+     * @return success with result containing relevant file information, error otherwise
+     */
+    suspend fun getFileMetadata(fullPath: String): Result<IONFILEMetadataResult> =
+        withContext(Dispatchers.IO) {
+            runCatching {
+                val file = File(fullPath)
+                if (!file.exists()) {
+                    throw IONFILEExceptions.DoesNotExist()
+                }
+                getMetadata(fileObject = file)
+            }
+        }
+
+    /**
+     * Create a file
+     *
+     * @param fullPath full path to the file to create
+     * @param options options to create the file
+     * @return success if the file was created successfully, error otherwise
+     */
+    suspend fun createFile(fullPath: String, options: IONFILECreateOptions): Result<Unit> =
+        withContext(Dispatchers.IO) { createDirOrFile(fullPath, options, isDirectory = false) }
+
+    /**
+     * Saves data to a file
+     *
+     * @param fullPath full path to the file to save
+     * @param options options to save the file
+     * @return success if the file was saved successfully, error otherwise
+     */
+    suspend fun saveFile(
+        fullPath: String,
+        options: IONFILESaveOptions
+    ): Result<Unit> = withContext(Dispatchers.IO) {
+        runCatching {
+            val file = File(fullPath)
+            if (!file.exists()) {
+                val createFileResult = createFile(
+                    fullPath,
+                    IONFILECreateOptions(recursive = options.createFileRecursive)
+                )
+                createFileResult.exceptionOrNull()?.let { throw it }
+            }
+            val fileStream = FileOutputStream(file, options.mode == IONFILESaveMode.APPEND)
+            if (options.encoding is IONFILEEncoding.WithCharset) {
+                val writer =
+                    BufferedWriter(OutputStreamWriter(fileStream, options.encoding.charset))
+                writer.use { writer.write(options.data) }
+            } else {
+                val outputStream = BufferedOutputStream(fileStream)
+                val dataToDecode = if (options.data.contains(",")) {
```

Comment:
Could this be refactored to the following?

`val dataToDecode = options.data.substringAfter(",", options.data)`